// @cucumber/cucumber/lib/formatter/get_color_fns.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
function getColorFns(enabled) {
    if (enabled) {
        return {
            forStatus(status) {
                return {
                    AMBIGUOUS: chalk_1.default.red.bind(chalk_1.default),
                    FAILED: chalk_1.default.red.bind(chalk_1.default),
                    PASSED: chalk_1.default.green.bind(chalk_1.default),
                    PENDING: chalk_1.default.yellow.bind(chalk_1.default),
                    SKIPPED: chalk_1.default.cyan.bind(chalk_1.default),
                    UNDEFINED: chalk_1.default.yellow.bind(chalk_1.default),
                    UNKNOWN: chalk_1.default.yellow.bind(chalk_1.default),
                }[status];
            },
            location: chalk_1.default.gray.bind(chalk_1.default),
            tag: chalk_1.default.cyan.bind(chalk_1.default),
            diffAdded: chalk_1.default.green.bind(chalk_1.default),
            diffRemoved: chalk_1.default.red.bind(chalk_1.default),
            errorMessage: chalk_1.default.red.bind(chalk_1.default),
            errorStack: chalk_1.default.grey.bind(chalk_1.default),
        };
    }
    else {
        return {
            forStatus(_status) {
                return (x) => x;
            },
            location: (x) => x,
            tag: (x) => x,
            diffAdded: (x) => x,
            diffRemoved: (x) => x,
            errorMessage: (x) => x,
            errorStack: (x) => x,
        };
    }
}
exports.default = getColorFns;
